lib/activerecord/delay_touching.rb

```diff
@@ -87,17 +87,23 @@ def self.touch_records(attr, klass, records)
           changes[column] = current_time
           records.each do |record|
             next if record.destroyed?
+            next unless record.persisted? # A new record may have been rolled back
             record.instance_eval do
               write_attribute column, current_time
               @changed_attributes.except!(*changes.keys)
             end
           end
         end
 
-        klass.unscoped.where(klass.primary_key => records).update_all(changes)
+        updatable_records = records.reject{|r| r.persisted?} # No need to touch unpersisted records
```

Comment:
Could we just cleanup the `records` array at the top of the method instead of trying to deal with it in multiple locations?